src/catkin_pkg/changelog_generator.py

```diff
@@ -168,6 +172,7 @@ def update_changelog_file(data, tag2log_entries, vcs_client=None, skip_contribut
             data = prepend_version_content(data, tag.name, content)
             assert data is not None
         else:
+            next_tag = None
```

Comment:
This doesn't seem sufficient since in line 185 `next_tag.name` is being used which would result in an exception when `next_tag` is `None`.

Reply:
I see. How about now?

Latest example:
```
# mkdir -p catkin_ws/src && cd catkin_ws/src
# git clone https://github.com/130s/ros_comm -b l/catkingpkg_207
:
Checking connectivity... done.
# cd ros_comm/
root@9edfee90dd08:~/catkin_ws/src/ros_comm# catkin_generate_changelog                          
Found packages: ros_comm
Querying commit information since latest tag...
Updating forthcoming section of changelog files...
- updating 'ros_comm/CHANGELOG.rst'
Traceback (most recent call last):
  File "/usr/bin/catkin_generate_changelog", line 11, in <module>
    load_entry_point('catkin-pkg==0.4.3', 'console_scripts', 'catkin_generate_changelog')()
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/cli/generate_changelog.py", line 130, in main_catching_runtime_error
    main(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/cli/generate_changelog.py", line 123, in main
    update_changelogs(base_path, packages_with, tag2log_entries, logger=logging, vcs_client=vcs_client, skip_contributors=args.skip_contributors)
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/changelog_generator.py", line 124, in update_changelogs
    data = update_changelog_file(data, pkg_tag2log_entries, vcs_client=vcs_client, skip_contributors=skip_contributors)
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/changelog_generator.py", line 180, in update_changelog_file
    raise RuntimeError('Could not find section "%s"' % next_tag.name)
UnboundLocalError: local variable 'next_tag' referenced before assignment
root@9edfee90dd08:~/catkin_ws/src/ros_comm# export PYTHONPATH=/root/catkin_ws/src/catkin_pkg/src:$PYTHONPATH

# git clone https://github.com/130s/catkin_pkg -b better_msg_142
:
Checking connectivity... done.

root@9edfee90dd08:~/catkin_ws/src/ros_comm# catkin_generate_changelog                          
Found packages: ros_comm
Querying commit information since latest tag...
ERROR: The tag name 'v1.13.6_rc1' doesn't match the version pattern x.y.z
```

Reply:
Neither of the two changes related to `next_tag` are relevant for this test case. The thing which makes it "work" is the added `raise RuntimeError`.